src/highlight_paths_spline.js

```diff
@@ -0,0 +1,326 @@
+        edge_on_paths = {};
+    }
+
+    function contains_path(paths) {
+        return function(path) {
+            return paths.indexOf(path)>=0;
+        };
+    }
+
+    // sigh
+    function doesnt_contain_path(paths) {
+        var cp = contains_path(paths);
+        return function(path) {
+            return !cp(path);
+        };
+    }
+
+    function intersect_paths(pathsA, pathsB) {
+        if(!pathsA || !pathsB)
+            return false;
+        return pathsA.some(contains_path(pathsB));
+    }
+
+    function toggle_paths(pathsA, pathsB) {
+        if(!pathsA)
+            return pathsB;
+        else if(!pathsB)
+            return pathsA;
+        if(pathsB.every(contains_path(pathsA)))
+            return pathsA.filter(doesnt_contain_path(pathsB));
+        else return pathsA.concat(pathsB.filter(doesnt_contain_path(pathsA)));
+    }
+
+    function add_behavior(chart, node, edge, ehover) {
+        chart
+            .cascade(300, true, conditional_properties(function(n) {
+                return intersect_paths(node_on_paths[chart.nodeKey.eval(n)], selected);
+            }, function(e) {
+                return intersect_paths(edge_on_paths[chart.edgeKey.eval(e)], selected);
+            }, selectprops));
+
+        node
+            .on('mouseover.highlight-paths', function(n) {
+                highlight_paths_group.hover_changed(node_on_paths[chart.nodeKey.eval(n)] || null);
+            })
+            .on('mouseout.highlight-paths', function(n) {
+                highlight_paths_group.hover_changed(null);
+            })
+            .on('click.highlight-paths', function(n) {
+                highlight_paths_group.select_changed(toggle_paths(selected, node_on_paths[chart.nodeKey.eval(n)]));
+            });
+
+    }
+
+    function remove_behavior(chart, node, edge, ehover) {
+        node
+            .on('mouseover.highlight-paths', null)
+            .on('mouseout.highlight-paths', null)
+            .on('click.highlight-paths', null);
+        clear_all_highlights();
+        chart
+            .cascade(300, false, selectprops)
+            .cascade(400, false, hoverprops);
+    }
+
+    highlight_paths_group
+        .on('hover_changed.highlight-paths-spline', function(hpaths) {
+            if(hoverpaths !== hpaths) {
+                hoverpaths = hpaths;
+                draw_hovered();
+            }
+        });
+
+    var _behavior = dc_graph.behavior('highlight-paths-spline', {
+        add_behavior: add_behavior,
+        remove_behavior: function(chart, node, edge, ehover) {
+            remove_behavior(chart, node, edge, ehover);
+            return this;
+        },
+        parent: function(p) {
+            if(p)
+                _anchor = p.anchorName();
+            // else we should have received anchor earlier
+            highlight_paths_group.on('paths_changed.' + _anchor, p ? paths_changed : null);
+            highlight_paths_group.on('select_changed.' + _anchor, p ? select_changed : null);
```

Comment:
the `select_changed` call isn't doing anything because highlight_paths does this already

maybe in the future we could consider weighting the selected path a lot in layout